File: content/en/04-llms/llms/_solutions/12_plot-image-1/12_plot-image-1.R
library(readr)
library(ellmer)
library(ggplot2)

mtcars <- read_csv(here::here("data/mtcars.csv"))

ggplot(mtcars) +
  aes(x = wt, y = mpg) +
  geom_point(color = "steelblue", size = 3) +
  labs(
    title = "MPG vs Weight",
    x = "Weight (1000 lb)",
    y = "Miles per Gallon (mpg)"
  ) +
  theme_bw()

chat <- chat_posit(model = "zai-org/GLM-5.3-Flash", echo = "output")
chat$chat(
  "Interpret this plot.",
  content_image_plot()
)

File: content/en/04-llms/llms/_solutions/12_plot-image-1/12_plot-image-1.py
# %%
import chatlas
import polars as pl
from matplotlib import pyplot as plt
from plotnine import aes, geom_point, ggplot, labs, theme_bw
from pyhere import here

# %%
mtcars = pl.read_csv(here("data/mtcars.csv"))
mtcars

# %%
p = (
    ggplot(mtcars, aes(x="wt", y="mpg"))
    + geom_point(color="steelblue", size=2)
    + labs(title="MPG vs Weight", x="Weight (1000 lb)", y="Miles per Gallon (mpg)")
    + theme_bw()
)

p.show()

# %%
# Register the plot with matplotlib's current figure
plt.figure(p.draw())

chat = chatlas.ChatPosit(model="zai-org/GLM-5.3-Flash")
chat.chat(
    "Interpret this plot.",
    chatlas.content_image_plot(),
)
